avocado/utils/pci.py

```diff
@@ -113,6 +113,25 @@ def get_nics_in_pci_address(pci_address):
     return net_interfaces_list
 
 
+def get_pci_class_name(pci_address):
+    """
+    Gets pci class name for given pci bus address
+
+    e.g: >>> pci.get_pci_class_name("0000:01:00.0")
+             'scsi_host'
+
+    :param pci_address: Any segment of a PCI address(1f, 0000:00:if, ...)
+
+    :return: class name for corresponding pci bus address
+    """
+    pci_class_dic = {'0104': 'scsi_host', '0c04': 'fc_host',
+                     '0200': 'net', '0108': 'nvme'}
+    pci_class_id = get_pci_prop(pci_address, "Class")
+    if pci_class_id not in pci_class_dic.keys():
```

Comment:
the `keys` is slow and unnecessary here, please use `if pci_class_id not in pci_class:`

Note: The `dict.keys()` actually creates a new list, therefor is really slow and should only be used when the dict is changing. To iterate through key/values/items the `iterkeys/itervalues/iteritems` is a better choice and the default `__in__` for dict actually uses `iterkeys`.